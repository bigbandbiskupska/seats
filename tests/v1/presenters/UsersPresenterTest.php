<?php

use App\Tests\BaseTestCase;
use Nette\Application\BadRequestException;
use Nette\Application\Request;
use Nette\Application\Responses\JsonResponse;
use Nette\Application\UI\Presenter;
use Nette\Http\IResponse;
use Nette\Utils\DateTime;
use Tester\Assert;

$container = require __DIR__ . "/../../bootstrap.php";

class UsersPresenterTest extends BaseTestCase {

    /** @var Presenter */
    protected $presenter;

    public function setUp() {
        $this->setUpRequestInput(array(
            'name' => 'New',
            'surname' => 'User',
            'email' => 'new@user.com',
            'password' => 'yyy',
            'ip_address' => '7.7.7.7',
            'expires_at' => DateTime::from("2017-01-01 00:00:00")
        ));

        $this->presenter = $this->createPresenter('v1:Users');
    }

    public function testUnauthorizedCreate() {
        Assert::exception(function() {
            $request = new Request('v1:Users', 'POST', array('action' => 'create'));
            $this->presenter->run($request);
        }, BadRequestException::class, null, IResponse::S401_UNAUTHORIZED);
    }

    public function testForbiddenCreate() {
        Assert::exception(function() {
            $request = new Request('v1:Users', 'POST', array('action' => 'create', 'token' => '1234'));
            $this->presenter->run($request);
        }, BadRequestException::class, null, IResponse::S403_FORBIDDEN);
    }

    public function testCreate() {

        $request = new Request('v1:Users', 'POST', array('action' => 'create', 'token' => 'abcd'));
        /** @var JsonResponse */
        $response = $this->presenter->run($request);
        $httpResponse = $this->container->getByType(IResponse::class);

        Assert::type(JsonResponse::class, $response);
        Assert::equal(IResponse::S200_OK, $httpResponse->getCode());
        Assert::true($response->getPayload() != null);

        $expected = [
            'id' => 4,
            'name' => 'New',
            'surname' => 'User',
            'email' => 'new@user.com',
            'password' => 'yyy',
            'token' => '',
            'roles' => 'user',
            'expires_at' => DateTime::from("2017-01-01 00:00:00"),
            'ip_address' => '7.7.7.7',
        ];
        $actual = $response->getPayload();
        Assert::equal($expected, $actual);

        Assert::count(2, $this->database->table('allowed_limit')->where('user_id', 4));
        foreach ($this->database->table('allowed_limit')->where('user_id', 4) as $limit) {
            Assert::equal(5, $limit->limit);
        }
    }

}

# Spuštění testovacích metod
run(new UsersPresenterTest($container));
