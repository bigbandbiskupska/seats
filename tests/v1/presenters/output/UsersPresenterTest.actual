[
	'id' => 4,
	'name' => 'New',
	'surname' => 'User',
	'email' => 'new@user.com',
	'password' => 'yyy',
	'roles' => 'user',
	'token' => '',
	'expires_at' => Nette\Utils\DateTime::__set_state(/* #5663 */ [
		'date' => '-0001-11-30 00:00:00.000000',
		'timezone_type' => 3,
		'timezone' => 'Europe/Prague',
	]),
	'ip_address' => '7.7.7.7',
]
