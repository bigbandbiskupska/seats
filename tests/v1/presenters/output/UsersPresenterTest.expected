[
	'id' => 4,
	'name' => 'New',
	'surname' => 'User',
	'email' => 'new@user.com',
	'password' => 'yyy',
	'token' => '',
	'ip_address' => '7.7.7.7',
]
